Improve discoverability of contribution/abstract types

nop33 opened the pull request:
![image](https://cloud.githubusercontent.com/assets/1579899/19516429/31e5e2e0-95fc-11e6-9031-f1f31d4d2172.png)

![image](https://cloud.githubusercontent.com/assets/1579899/19516445/450f4eec-95fc-11e6-97c5-d52229e85a6d.png)

ThiefMaster commented:
We don't use the term "abstract type" anywhere; not sure if it's a good fit.

It's was called "Presentation type" in two places (legacy) and "Contribution type" everywhere else.

ThiefMaster commented:
I think it looks way too intrusive on the contributions page now. It's something you setup once and then don't care about it anymore when managing contributions etc.

nop33 commented:
We might not call them `abstract types` anywhere, but we get many tickets with people looking for the `abstract types`, `presentation types`, `contribution types`... In any case, it's the `type` you choose when you submit an abstract, so personally I would go with `abstract type`. What do the rest think?

nop33 commented:
@ThiefMaster it might be that you configure it once, but if you don't find it at all then you cannot configure it.

OmeGak commented:
Contribution type all the way, I would say.

nop33 commented:
## Updated
![image](https://cloud.githubusercontent.com/assets/1579899/19518204/375b8a7c-9606-11e6-9d2b-12969bb501cd.png)

ThiefMaster commented:
Hm.. why link to the paeg and not just show the dialog?

nop33 commented:
So that the people are informed that that's where they need to go to configure the contribution types. If we open the dialog, they will always rely on finding the contribution types in the abstracts page.

ThiefMaster commented:
To me that feels like making the UX worse just to educate users - and in the end I think it's perfectly fine to configure it from there; I don't think there are that many cases where people use different contribution types while not using abstracts.

pferreir commented:
I agree with @thiefmaster: contribution types, despite their name, are basically useless outside the abstract reviewing mode.

nop33 commented:
OK then! What's the reason then of having them in the contribution page? The naming?

pferreir commented:
The name, I believe. Anyway, they're kind of hidden which is good. The main entry point should be the CFA setup page and users who try to guess where they are based on the name should be able to get there too.

pferreir commented:
Why are we opening this ticket? Has there been a user complaint?

ThiefMaster commented:
INC1165676

nop33 commented:
It's not the first time I get contact by Lionel or Hakan about this problem. It was a quick fix and I created a PR for it.

pferreir commented:
Sure, but there are 2 things here:
 1. It may be annoying but from what I see it's not urgent;
 2. We have a new release about to come out that makes that dialog accessible from what should be its main entry point.

That said, does it really make sense to go for a PR that substantially changes the way users perceive this feature?

ThiefMaster commented:
I'd keep the cog menu in contributions as it is and open the dialog from the abstracts menu...

nop33 commented:
I'll add the dialog in the CFA page.

nop33 commented:
### Update

What about this approach?
1. I added a page description and the text "Settings" next to the cog icon in the contribution page, to help users find the settings for the **Contribution types** and **Custom fields** easier.
2. In the CFA page, I created a new section with the title **Fields and types** where I added the buttons for managing the **Contribution types** and **Abstract fields**.
3. I improved the UI in the `.sortable-list` adding the option to make the items look like `.tiles`.


![image](https://cloud.githubusercontent.com/assets/1579899/20058441/5ddabe3a-a4f1-11e6-88f6-de62051c05ff.png)

![image](https://cloud.githubusercontent.com/assets/1579899/20058434/4b7425f6-a4f1-11e6-9774-a2890bdfc3da.png)

![image](https://cloud.githubusercontent.com/assets/1579899/20058454/6d2da80c-a4f1-11e6-9b87-48fa70136245.png)

OmeGak commented:
Not a strong opinion. It's only that "Fields and types" feels to me like a bit out of place there.

nop33 commented:
I think the title "Fields and types" is helping users discover exactly what they are looking for.

OmeGak commented:
Yes, I just said that it feels a bit out of place. Dunno how we could make it look better. Maybe not in the same group as `Submission`/`Reviewing`/`BoA`? Standalone like `List of abstracts`?

nop33 commented:
@OmeGak IMO it should be close to the `Submission` (it should actually be part of it, but we cannot fit too many buttons there), so I wouldn't put it in a separate group. It's part of the "Settings" group, whereas the "List of abstracts" is not related to settings. Moving it outside of the group wouldn't make any semantic sense.

ThiefMaster commented on indico/htdocs/sass/widgets/_sortable_list.scss:
```diff
@@ -132,5 +132,24 @@
-   }
+    }
+
+    &.tiles{
```
missing space before `{`

ThiefMaster commented on indico/modules/events/abstracts/forms.py:
```diff
@@ -413,8 +413,8 @@ class CreateEmailTemplateForm(EditEmailTemplateRuleForm):
     title = StringField(_("Title"), [DataRequired()])
-    description = IndicoMarkdownField(_('Content'), editor=True, mathjax=True)
-    submitted_contrib_type = QuerySelectField(_("Type"), get_label='name', allow_blank=True,
+    description = IndicoMarkdownField(_('Content'), [DataRequired()], editor=True, mathjax=True)
```
no...

ThiefMaster commented on indico/modules/events/abstracts/forms.py:
```diff
@@ -413,8 +413,8 @@ class CreateEmailTemplateForm(EditEmailTemplateRuleForm):
-    description = IndicoMarkdownField(_('Content'), editor=True, mathjax=True)
-    submitted_contrib_type = QuerySelectField(_("Type"), get_label='name', allow_blank=True,
+    description = IndicoMarkdownField(_('Content'), [DataRequired()], editor=True, mathjax=True)
+    submitted_contrib_type = QuerySelectField(_("Contribution type"), get_label='name', allow_blank=True,
```
not sure if that rename is so useful. "Type: Oral/Poster" etc is pretty clear already

other opinions?

nop33 replied:
I think we made it clear when I opened the pull request that we want to rename it to "Contribution type" because in legacy it was "Presentation type". It also makes the transition from an abstract to a contribution easier for the user.

ThiefMaster commented on indico/modules/events/contributions/templates/management/contributions.html:
```diff
@@ -10,16 +10,16 @@
         <a id="contributions-settings-dropdown"
            class="i-button borderless icon-settings arrow js-dropdown"
-           data-toggle="dropdown"></a>
+           data-toggle="dropdown">Settings</a>
```
Looks a bit weird to me (also, missing i18n)...

![](https://cloud.githubusercontent.com/assets/179599/20703331/9a4888a8-b61b-11e6-9faf-a679541eab80.png)

nop33 replied:
For me it looks better like this than with just the cog icon.

ThiefMaster commented on indico/modules/events/contributions/templates/management/contributions.html:
```diff
@@ -36,6 +36,12 @@
 
+{% block description %}
+    {%- trans -%}
+        Manage the contributions of the event from the list below or configure the <strong>contribution types</strong> and <strong>custom fields</strong> from the settings menu above.
```
break line after `</strong>`?

ThiefMaster commented on indico/modules/events/abstracts/templates/management/overview.html:
```diff
@@ -14,33 +14,53 @@
+                       data-href="{{ url_for('contributions.manage_types', event) }}"
+                       data-title="{% trans %}Manage contribution types{% endtrans %}"
+                       data-ajax-dialog>
+                        {%- trans %}Contribution types{% endtrans -%}
```
Those are the only buttons on the page with no icon now :/

nop33 replied:
Not the only ones ("Close" call for abstracts also doesn't), but I believe we're overusing icons... Any better idea?

nop33 commented on indico/modules/events/abstracts/views.py:
```diff
@@ -45,7 +45,8 @@ def getCSSFiles(self):
                 self._asset_env['selectize_css'].urls() +
-                self._asset_env['abstracts_sass'].urls())
+                self._asset_env['abstracts_sass'].urls() +
+                self._asset_env['contributions_sass'].urls())
```
It's for the "Contribution types" dialog. You really think I should move the code to `eventmanagement.scss`? We also import the file in `WPDisplayAbstractsBase`, not just `WPManageAbstracts`...

ThiefMaster replied:
OK, nevermind then
